### PR TITLE
Update .gitignore.

### PR DESCRIPTION
**Pull request by domob1812**
Update the top-level .gitignore to fit the generated files I see on a
GNU/Linux system more closely.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-src/bitcoind
 .*.swp
 *.*~*
+qrc_bitcoin.cpp
+namecoin-qt
+namecoin-qt.exe
 src/namecoind
+src/namecoind.exe
+build
+Makefile
```
Comment (belovachap) [inline .gitignore]: Don't we want to track the Makefile?

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,10 @@
-src/*.exe
 *.pyc
-src/bitcoin
-src/bitcoind
 .*.swp
 *.*~*
+qrc_bitcoin.cpp
```
Comment (belovachap) [inline .gitignore]: What's qrc_bitcoin.cpp and why do we want to ignore it?

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
+namecoin-qt
+namecoin-qt.exe
 src/namecoind
+src/namecoind.exe
+build
```
Comment (belovachap) [inline .gitignore:11]: Not familiar with the build process, what command do you run that creates a build directory?

File: .gitignore (hunk 1 above)
Comment (domob1812) [inline .gitignore]: No, the (top-level) Makefile is created by qmake.

**Comment by domob1812** [comment]
I'm also not familiar with the build system (Qt) very much ... maybe some of these are relicts of older versions.  It would be cool if someone especially on other systems could test further (or with more Qt experience could chim in).  The Makefile is definitely generated by Qt, though (see the top lines), and the namecoin-qt binary should be ignored, too.

**Comment by belovachap** [comment]
Ah, I think I see what you're trying to do. A couple of thoughts: 1) Makefile will ignore Makefiles in all directories (probably want `/Makefile` if we just want to ignore the top level one; http://stackoverflow.com/questions/3637660/how-to-exclude-file-only-from-root-folder-in-git). It kind of works as-is because the Makefile in src/ is already being tracked. 2) Gathering together `/qrc_bitcoin.cpp`, `build/`, and `/Makefile` under a comment like `# Qt build crud` might give a good hint about why these are in the .gitignore file.

**Comment by domob1812** [comment]
Ok, I've updated the pull request accordingly.

**Comment by belovachap** [comment]
Looks good to me! :)

**Pull request merged**